crates/astria-sequencer-relayer/src/data_availability.rs

```diff
@@ -9,6 +12,13 @@ use astria_celestia_jsonrpc_client::{
     Client,
     ErrorKind,
 };
+use astria_sequencer::proposal::{
```

Comment:
Early feedback on this: we should not increase coupling between sequencer and sequencer-relayer further, but split this out into a free-standing crate, and instead have something like `astria-sequencer-core`.

Also `InclusionProof` and `MerkleTree` are used in RPCs/as part of a public API, so should we make these into protobufs?

Reply:
yes agreed, going to put all this shared stuff into its own crate